### PR TITLE
Added SNMPv3 user to render in output

### PR DESCRIPTION
**Pull request by roverflow**
##### SUMMARY
This pull request addresses an issue where SNMPv3 user configuration was not being displayed in the running-config. The fix involves adding a new command to retrieve the output and parse it to add it to the facts list

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME

- plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py

### CODE REVIEW

**Comment by KB-perByte** [comment]
Let's add some unit tests to this change!

**Comment by KB-perByte** [comment]
@roverflow if you could also update the examples as per https://github.com/ansible-collections/cisco.ios/pull/786
it would be helpful! 
Regards

File: tests/unit/modules/network/ios/test_ios_snmp_server.py [hunk 1]
```diff
@@ -64,6 +71,7 @@ def tearDown(self):
         self.mock_get_config.stop()
         self.mock_load_config.stop()
         self.mock_execute_show_command.stop()
+        self.mock_execute_show_command_user.stop()
 
     def test_ios_snmp_server_merged_idempotent(self):
```
Comment (KB-perByte) [inline tests/unit/modules/network/ios/test_ios_snmp_server.py:76]: Please add tests for action states

File: changelogs/fragments/snmpv3_user_render.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - SNMPv3 user rendering now works correctly.
```
Comment (KB-perByte) [inline changelogs/fragments/snmpv3_user_render.yml:3]: A bit more descriptive here!

File: plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py [hunk 3]
```diff
@@ -57,7 +62,26 @@ def host_traps_string_to_list(self, hosts):
                     element["traps"] = list(element.get("traps").split())
             return hosts
 
-    def populate_facts(self, connection, ansible_facts, data=None):
+    def get_snmpv3_user_facts(self, snmpv3_user):
```
Comment (KB-perByte) [inline plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py:65]: As this, is not a trivial resource module way, can we add some docstrings and comment to the code for us to reference it better later

Comment (KB-perByte) [reply plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py:65]: Add a note to the user attribute in docs to state that the behavior is supposed to impact idempotency.

File: plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py [hunk 4]
```diff
@@ -73,11 +113,21 @@ def populate_facts(self, connection, ansible_facts, data=None):
...
+            snmpv3_user = self.get_snmpv3_user_data(connection)
 
         # parse native config using the Snmp_server template
         snmp_server_parser = Snmp_serverTemplate(lines=data.splitlines(), module=self._module)
+        # parse snmpv3_user data using the get_snmpv3_user_facts method
+        snmp_user_data = self.get_snmpv3_user_facts(snmpv3_user)
         objs = snmp_server_parser.parse()
 
+        # add snmpv3_user data to the objs dictionary
+        if len(snmp_user_data) != 0:
```
Comment (KB-perByte) [inline plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py:126]: this condition can be simplified

File: changelogs/fragments/snmpv3_user_render.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - Adding SNMPv3 user config data using different command in addition to running-config.
```
Comment (KB-perByte) [inline changelogs/fragments/snmpv3_user_render.yml:3]: this needs to be corrected

**Review by KB-perByte — CHANGES_REQUESTED** [review]

File: plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py [hunk 6]
```diff
@@ -73,11 +113,21 @@ def populate_facts(self, connection, ansible_facts, data=None):
...
+            snmpv3_user = self.get_snmpv3_user_data(connection)
 
         # parse native config using the Snmp_server template
         snmp_server_parser = Snmp_serverTemplate(lines=data.splitlines(), module=self._module)
+        # parse snmpv3_user data using the get_snmpv3_user_facts method
+        snmp_user_data = self.get_snmpv3_user_facts(snmpv3_user)
         objs = snmp_server_parser.parse()
 
+        # add snmpv3_user data to the objs dictionary
+        if len(snmp_user_data):
```
Comment (KB-perByte) [inline plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py:126]: len() is also not required

Comment (roverflow) [reply plugins/module_utils/network/ios/facts/snmp_server/snmp_server.py:126]: Got it!

File: changelogs/fragments/snmpv3_user_render.yml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - snmp_server - updated module to get snmpv3 user data from different command.
```
Comment (KB-perByte) [inline changelogs/fragments/snmpv3_user_render.yml:3]:
```suggestion
  - snmp_server - update module to get snmp_server user configuration.
```

**Review by KB-perByte — APPROVED** [review]

**Pull request merged**